lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java

```diff
@@ -9208,4 +9295,49 @@ public String addTwitterTimeline(){
 	public boolean isGradebookExists() {
 		return (this.getCurrentTool(this.GRADEBOOK_TOOL_ID) != null || this.getCurrentTool(this.GRADEBOOK_CLASSIC_TOOL_ID) != null);
 	}
+	
+	
+	private void updateTask(String reference, SimplePage page, Date dueDate) {
+		updateTask(reference, page.getTitle(), dueDate);
+	}
+	
+	private void createTask(String reference, SimplePage page, Date dueDate) {
+		createTask(reference, page.getSiteId(), page.getTitle(), dueDate);
+	}
+	
+	private void updateTask(String reference, String title, Date dueDate) {
```

Comment:
createOrUpdateTask will do this work and this will not be needed, am I right?

Reply:
Yes, we need this one because sometimes we need to update a task that has been previously created, but the user doesn't want to create a new task.